taf/TAF_Runner.py

```diff
@@ -889,6 +894,59 @@ def audit(self):
             self.logger.info('No valid Spark Session')
             return None
 
+    # ----------------------------------------------------
+    # Should be called from the runner due to extra step
+    # ----------------------------------------------------
+    def prep_meta_table(self,
+                        pgm_name: str,
+                        step_name: str,
+                        filetyp: str = None,
+                        fl: str = None,
+                        fl2: str = None,
+                        rpt_out: str = None,
+                        clm_tbl: str = None,
+                        bsf_file_date: str = None):
+        # consider putting logging here and explain why value is needed.
+        if pgm_name is None:
+            pass
+
+        spark = SparkSession.getActiveSession()
+        if spark is not None:
+            pgmLkpDF = (spark.createDataFrame(data=TAF_Metadata.pgmLkpData, schema=TAF_Metadata.pgmLkpSchema))
+            pgmLkpDF.createOrReplaceTempView("audt_lkup_unfmt")
+            self.__create_formated_lkup(filetyp, fl, fl2, rpt_out, clm_tbl, bsf_file_date)
+
+        self.__getcounts(pgm_name=pgm_name, step_name=step_name)
+
+    # ---------------------------------------------------------
+    # private function to lookup and return the answer
+    # ---------------------------------------------------------
+    def __create_formated_lkup(self, filetyp: str, fl: str, fl2: str, rpt_out: str, clm_tbl: str, bsf_file_date: str):
+        from pyspark.sql import DataFrame
+        import pandas as pd
+
+        data: list = []
+        fmted_lookup_df: pd.DataFrame = pd.DataFrame(data=data, columns=TAF_Metadata.pgmLkpSchema)
+        spark = SparkSession.getActiveSession()
+
+        if spark is not None:
+            col_num = 0
+            for row in TAF_Metadata.pgmLkpData:
```

Comment:
not sure what this is but i recommend to use spark for better performance than map reduce/hadoop

Reply:
Yes. Hopefully at some point we'll be able to clean up and optimize the code once our timelines free up.